content/300-guides/100-prisma-guides/100-prisma-client-transactions-guide.mdx

```diff
@@ -0,0 +1,778 @@
+
+```ts
+await client.email.updateMany({
+  where: {
+    user: {
+      id: 10,
+      unread: true
+    }
+  },
+  data: {
+    unread: false
+  }
+})
+```
+
+#### Can I use nested writes with bulk operations?
+
+No - neither `updateMany` nor `deleteMany` currently supports nested writes. For example, you cannot delete multiple teams and all of their members (a cascading delete):
+
+```ts highlight=8;delete
+  const deleteMany = await prisma.team.deleteMany({
+    where: {
+      id: {
+        in: [2, 99, 2, 11]
+      }
+    },
+    data: {
+      members: { } // Cannot access members here
+    },
+  })
+```
+
+You an use the [`$transaction` API to perform a cascading delete](#scenario-cascading-delete).
+
+#### Can I use bulk operations with the `$transaction` API?
+
+Yes - for example, you can include multiple `deleteMany` operations inside a `$transaction`.
+
+### <inlinecode>$transaction</inlinecode> API
+
+The `$transaction` API is generic solution to independent writes that allows you to run multiple operations as a single, atomic operation - if any operation fails, Prisma rolls back the entire transaction. As Prisma Client evolves, use cases for the `$transaction` API will increasingly be replaced by more specialized bulk operations (such as `createMany`) and nested writes.
+
+#### When to use the `$transaction` API
+
+Consider the `$transaction` API if:
+
+* ✔ You want to update a batch that includes different types of records, such as emails and users. The records do not need to be related in any way.
+* ✔ You want to batch raw SQL queries (`$executeRaw`) - for example, for features that Prisma Client does not yet support.
+
+#### Scenario: Privacy legislation
+
+GDPR and other privacy legislation give users the right to request that an organization deletes all of their personal data. In the following example schema, a `User` can have many posts and private messages:
+
+```prisma
+model User {
+  id                Int       @id @default(autoincrement())
+  posts             Post[]
+  privateMessages   PrivateMessage[]
+}
+
+model Post {
+  id         Int     @id @default(autoincrement())
+  user       User    @relation(fields: [userId], references: [id])
+  userId     Int  
+  title      String
+  content    String
+}
+
+model PrivateMessage {
+  id         Int     @id @default(autoincrement())
+  user       User    @relation(fields: [userId], references: [id])
+  userId     Int  
+  message      String
+}
+```
+
+If a user invokes the right to be forgotten, we must delete three records: the user record, private messages, and posts. It is critical that _all_ delete operations succeed together or not at all, which makes this a use case for a transaction. However, using a single bulk operation like `deleteMany` is not possible in this scenario because we need to delete across three models. Instead, we can use the `$transaction` API to run three operations together - two `deleteMany` and one `delete`:
```

Comment:
Ah, this is an interesting example because the models could be related (user's messages and posts), but we lack the capabilities to do this nested write because we don't support nested deleteMany's. 

Taking a mental note of this.

The way I've thought about $transaction so far is that it'll eventually only be for unrelated records. It's really hard to come up with a use case for operating on unrelated data, so I think this is fine to keep.